File: backend/app/schemas/career.py
"""Life Work Engine API schemas."""
from datetime import date, datetime
from uuid import UUID
from pydantic import BaseModel
from typing import Optional


# ----- Mission -----
class MissionCreate(BaseModel):
    title: str
    description: Optional[str] = None
    status: str = "active"
    phase: Optional[str] = None
    priority: Optional[int] = None
    target_date: Optional[date] = None


class MissionUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    phase: Optional[str] = None
    priority: Optional[int] = None
    target_date: Optional[date] = None


class MilestoneCreate(BaseModel):
    mission_id: UUID
    title: str
    description: Optional[str] = None
    status: str = "pending"
    order_index: int = 0


class MilestoneUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    completed_at: Optional[datetime] = None
    order_index: Optional[int] = None


class MilestoneResponse(BaseModel):
    id: UUID
    mission_id: UUID
    title: str
    description: Optional[str]
    status: str
    completed_at: Optional[datetime]
    order_index: int

    class Config:
        from_attributes = True


class MissionResponse(BaseModel):
    id: UUID
    user_id: int
    title: str
    description: Optional[str]
    status: str
    phase: Optional[str]
    priority: Optional[int]
    target_date: Optional[date]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]
    milestones: list[MilestoneResponse] = []

    class Config:
        from_attributes = True


# ----- Achievement -----
class AchievementCreate(BaseModel):
    title: str
    description: Optional[str] = None
    category: str  # business | technical | financial | social_professional | personal_growth
    impact_level: Optional[str] = None  # low | medium | high | transformative
    date: date
    proof_url: Optional[str] = None


class AchievementResponse(BaseModel):
    id: UUID
    user_id: int
    title: str
    description: Optional[str]
    category: str
    impact_level: Optional[str]
    date: date
    proof_url: Optional[str]
    created_at: Optional[datetime]

    class Config:
        from_attributes = True


# ----- Opportunity -----
class OpportunityCreate(BaseModel):
    title: str
    type: str  # freelance_lead | job | partnership | startup_idea | collaboration | investment
    source: Optional[str] = None
    status: str = "open"
    value_estimate: Optional[str] = None
    notes: Optional[str] = None


class OpportunityUpdate(BaseModel):
    title: Optional[str] = None
    type: Optional[str] = None
    source: Optional[str] = None
    status: Optional[str] = None
    value_estimate: Optional[str] = None
    notes: Optional[str] = None


class OpportunityResponse(BaseModel):
    id: UUID
    user_id: int
    title: str
    type: str
    source: Optional[str]
    status: str
    value_estimate: Optional[str]
    notes: Optional[str]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True


# ----- Career Leverage -----
class CareerLeverageCreate(BaseModel):
    area: str  # skills | reputation | network | assets_projects
    score: float = 0.0
    notes: Optional[str] = None


class CareerLeverageUpdate(BaseModel):
    score: Optional[float] = None
    notes: Optional[str] = None


class CareerLeverageResponse(BaseModel):
    id: UUID
    user_id: int
    area: str
    score: float
    notes: Optional[str]
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True


# ----- Energy Pattern -----
class EnergyPatternCreate(BaseModel):
    work_type: str
    energy_effect: str  # energizes | drains | neutral
    focus_quality: Optional[str] = None  # high | medium | low
    notes: Optional[str] = None


class EnergyPatternResponse(BaseModel):
    id: UUID
    user_id: int
    work_type: str
    energy_effect: str
    focus_quality: Optional[str]
    notes: Optional[str]
    recorded_at: Optional[datetime]

    class Config:
        from_attributes = True


# ----- Dashboard -----
class CareerDashboardResponse(BaseModel):
    missions: list[MissionResponse]
    achievements_recent: list[AchievementResponse]
    opportunities_by_status: dict[str, int]
    leverage: list[CareerLeverageResponse]
    weakest_leverage_area: Optional[str] = None
    recommended_leverage_action: Optional[str] = None
    energy_insights: list[str] = []

File: frontend/src/services/career.ts
import { api } from './api'

export interface Milestone {
  id: string
  mission_id: string
  title: string
  description?: string
  status: string
  completed_at?: string
  order_index: number
}

export interface Mission {
  id: string
  user_id: number
  title: string
  description?: string
  status: string
  phase?: string
  priority?: number
  target_date?: string
  created_at?: string
  updated_at?: string
  milestones: Milestone[]
}

export interface LifeWorkAchievement {
  id: string
  user_id: number
  title: string
  description?: string
  category: string
  impact_level?: string
  date: string
  proof_url?: string
  created_at?: string
}

export interface Opportunity {
  id: string
  user_id: number
  title: string
  type: string
  source?: string
  status: string
  value_estimate?: string
  notes?: string
  created_at?: string
  updated_at?: string
}

export interface CareerLeverage {
  id: string
  user_id: number
  area: string
  score: number
  notes?: string
  updated_at?: string
}

export interface EnergyPattern {
  id: string
  user_id: number
  work_type: string
  energy_effect: string
  focus_quality?: string
  notes?: string
  recorded_at?: string
}

export interface CareerDashboard {
  missions: Mission[]
  achievements_recent: LifeWorkAchievement[]
  opportunities_by_status: Record<string, number>
  leverage: CareerLeverage[]
  weakest_leverage_area?: string
  recommended_leverage_action?: string
  energy_insights: string[]
}

export const careerService = {
  getDashboard: () => api.get<CareerDashboard>('/career/dashboard'),
  listMissions: (status?: string) =>
    api.get<Mission[]>(`/career/missions${status ? `?status=${status}` : ''}`),
  getMission: (id: string) => api.get<Mission>(`/career/missions/${id}`),
  createMission: (data: { title: string; description?: string; status?: string; phase?: string; priority?: number; target_date?: string }) =>
    api.post<Mission>('/career/missions', data),
  updateMission: (id: string, data: Partial<Pick<Mission, 'title' | 'description' | 'status' | 'phase' | 'priority' | 'target_date'>>) =>
    api.patch<Mission>(`/career/missions/${id}`, data),
  deleteMission: (id: string) => api.delete(`/career/missions/${id}`),

  createMilestone: (data: { mission_id: string; title: string; description?: string; status?: string; order_index?: number }) =>
    api.post<Milestone>('/career/milestones', data),
  updateMilestone: (id: string, data: Partial<Pick<Milestone, 'title' | 'description' | 'status' | 'completed_at' | 'order_index'>>) =>
    api.patch<Milestone>(`/career/milestones/${id}`, data),
  deleteMilestone: (id: string) => api.delete(`/career/milestones/${id}`),

  listAchievements: (category?: string, limit?: number) => {
    const params = new URLSearchParams()
    if (category) params.set('category', category)
    if (limit != null) params.set('limit', String(limit))
    return api.get<LifeWorkAchievement[]>(`/career/achievements${params.toString() ? `?${params.toString()}` : ''}`)
  },
  createAchievement: (data: { title: string; description?: string; category: string; impact_level?: string; date: string; proof_url?: string }) =>
    api.post<LifeWorkAchievement>('/career/achievements', data),
  deleteAchievement: (id: string) => api.delete(`/career/achievements/${id}`),

  listOpportunities: (status?: string, type?: string) => {
    const params = new URLSearchParams()
    if (status) params.set('status', status)
    if (type) params.set('type_', type)
    return api.get<Opportunity[]>(`/career/opportunities${params.toString() ? `?${params.toString()}` : ''}`)
  },
  createOpportunity: (data: { title: string; type: string; source?: string; status?: string; value_estimate?: string; notes?: string }) =>
    api.post<Opportunity>('/career/opportunities', data),
  updateOpportunity: (id: string, data: Partial<Pick<Opportunity, 'title' | 'type' | 'source' | 'status' | 'value_estimate' | 'notes'>>) =>
    api.patch<Opportunity>(`/career/opportunities/${id}`, data),
  deleteOpportunity: (id: string) => api.delete(`/career/opportunities/${id}`),

  listLeverage: () => api.get<CareerLeverage[]>('/career/leverage'),
  updateLeverage: (area: string, data: { score?: number; notes?: string }) =>
    api.patch<CareerLeverage>(`/career/leverage/${area}`, data),

  listEnergyPatterns: () => api.get<EnergyPattern[]>('/career/energy-patterns'),
  createEnergyPattern: (data: { work_type: string; energy_effect: string; focus_quality?: string; notes?: string }) =>
    api.post<EnergyPattern>('/career/energy-patterns', data),
}
